File: atividade6.js


// Função para calcular o quadrado de um número
function quadrado(numero) {
    let resultado = Math.pow(numero, 2); // Variável local
    return resultado;
}
console.log("Quadrado de 5:", quadrado(5));

// Passagem por valor
function dobroValor(n) {
    return n * 2;
}

// Passagem por referência (simulada com array)
function dobroReferencia(nArray) {
    nArray[0] *= 2;
}

let num = 10;
console.log("Dobro por valor:", dobroValor(num));
console.log("Valor original após chamada:", num);

let numLista = [10];
dobroReferencia(numLista);
console.log("Dobro por referência:", numLista[0]);

// Simulando sobrecarga de funções usando parâmetros opcionais
function soma(a, b, c = null) {
    if (c === null) {
        return a + b;
    }
    return a + b + c;
}

// Função separada para soma de floats
function somaFloat(a, b) {
    return a + b;
}

console.log("Soma de dois inteiros:", soma(3, 4));
console.log("Soma de três inteiros:", soma(3, 4, 5));
console.log("Soma de dois floats:", somaFloat(3.5, 2.5));

File: atividade6.py
# Python

def quadrado(numero):
    resultado = numero ** 2  # Variável local
    return resultado

print("Quadrado de 5:", quadrado(5))

# Passagem por valor
def dobro_valor(n):
    return n * 2

# Passagem por referência (simulada com lista)
def dobro_referencia(n_lista):
    n_lista[0] *= 2

num = 10
print("Dobro por valor:", dobro_valor(num))
print("Valor original após chamada:", num)

num_lista = [10]
dobro_referencia(num_lista)
print("Dobro por referência:", num_lista[0])

# Sobrecarga de funções em Python (simulada usando argumentos padrões e *args)
def soma(a, b, c=None):
    if c is None:
        return a + b
    return a + b + c

def soma_float(a: float, b: float) -> float:
    return a + b

print("Soma de dois inteiros:", soma(3, 4))
print("Soma de três inteiros:", soma(3, 4, 5))
print("Soma de dois floats:", soma_float(3.5, 2.5))
